loadtest/control/simulcontroller/websocket.go

```diff
@@ -36,6 +36,12 @@ func (c *SimulController) wsEventHandler(wg *sync.WaitGroup) {
 				break
 			}
 
+			if ack, ok := ev.GetData()["should_ack"]; ok && ack.(bool) {
+				go func() {
```

Comment:
As I mentioned in channel I see this as a decent workaround but I'd prefer if we fixed the root issue rather than starting a pattern of avoiding the deadlock. My suggestion would be to make `ue.dataChan` a buffered channel and do a non-blocking write in `UserEntity.PostedAck()` with the default case (channel full) returning an error (to be logged).

Reply:
Couple questions:
1) How big of a buffer do we want?
2) If the buffer is full, will it just lose events? If so, this might need that tracking of ACKs might not be super accurate, which isn't necessarily a problem for load testing, but isn't a great simulation.

Reply:
So one thing to note, I think it's already sending it through the channel:
https://github.com/mattermost/mattermost-load-test-ng/pull/722/files#diff-d66de5fdcf416b8ffb207a36b55fca5b16554291b2a06a94e45d7372cc3272b3R307

Is the buffering something that would fix this? Given that it was dead-locking before, is calling this method causing a blocking call or something?

Reply:
Problem is that sending to `dataChan` is blocking right now and `dataChan` is unbuffered which causes the deadlock. We need to address both things.

Reply:
If we only make it non-blocking it will fail to send (chan full) as soon as there's a bit of extra concurrency. If we only make it buffered it could deadlock again if channel becomes full. I'd prefer to error than deadlock.